realm/realm-library/src/objectServer/java/io/realm/RealmFunctions.java

```diff
@@ -15,16 +15,30 @@
  */
 package io.realm;
 
-import org.bson.BsonValue;
+import org.bson.codecs.configuration.CodecRegistry;
 
 import io.realm.internal.jni.JniBsonProtocol;
 
-public class RealmFunctions {
+// FIXME This class is only a placeholder for JNI round trip as  until actual RealmFunctions
+//  implementation supersedes it.
+class RealmFunctions {
+
+    private CodecRegistry codecRegistry;
+
+    RealmFunctions(CodecRegistry codecRegistry) {
+        this.codecRegistry = codecRegistry;
+    }
+
+    // FIXME Prelimiry implementation to be able to test passing BsonValues through JNI
+    <T> T invoke(Object arg, Class<T> resultClass) {
```

Comment:
Rather than deleting these when the functions are ready, I would actually put them in their own `JniBsonProtocolTest` file. I think these two functions are super useful in and of themselves to make sure we never brake the conversion.

Reply:
Yes, it already pinpointed some errors in C++ parsing. I migrated the test to work across the actual functions by _echoing_  from the Stitch server in #6810 , as it seems like there is currently no way to easily do _test only_ JNI implementation.